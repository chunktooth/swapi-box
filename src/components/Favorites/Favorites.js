import React from 'react';
import Card from '../Card/Card';
import PropTypes from 'prop-types';

const Favorites = ({data, setFavorites}) => {
	const displayFavorites = data.map((favorite, index) => {
		return <Card 
			key={index}
			data={favorite}
			setFavorites={setFavorites} 
		/>
	})

	return (
		<article>
			{displayFavorites}
		</article>
	)
}

export default Favorites;