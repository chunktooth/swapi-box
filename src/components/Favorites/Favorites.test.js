import React from 'react';
import Favorites from './Favorites';
import {shallow} from 'enzyme';

describe('Favorites', () => {
	const mockFunction = jest.fn()
	const mockData = [{
		name: 'R2-D2',
		homeworld: 'Naboo',
		population: '300'
	}];
	const wrapper = shallow(<Favorites 
		data={mockData}
		setCategory={mockFunction}
		/>);	

	it('should match the snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	});

});