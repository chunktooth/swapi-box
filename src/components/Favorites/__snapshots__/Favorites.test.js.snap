// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Favorites
    data={
        Array [
            Object {
              "homeworld": "Naboo",
              "name": "R2-D2",
              "population": "300",
            },
          ]
    }
    setCategory={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          data={
                    Object {
                              "homeworld": "Naboo",
                              "name": "R2-D2",
                              "population": "300",
                            }
          }
          setFavorites={undefined}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "data": Object {
            "homeworld": "Naboo",
            "name": "R2-D2",
            "population": "300",
          },
          "setFavorites": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            data={
                        Object {
                                    "homeworld": "Naboo",
                                    "name": "R2-D2",
                                    "population": "300",
                                  }
            }
            setFavorites={undefined}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "data": Object {
              "homeworld": "Naboo",
              "name": "R2-D2",
              "population": "300",
            },
            "setFavorites": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
